src/topology/algebra/module/finite_dimension.lean

```diff
@@ -0,0 +1,332 @@
+    rw H,
+    exact continuous_zero },
+  { -- In the case where `l` is surjective, we factor it as `φ : (E ⧸ l.ker) ≃ₗ[𝕜] 𝕜`. Note that
+    -- `E ⧸ l.ker` is T2 since `l.ker` is closed.
+    have : finrank 𝕜 l.range = 1,
+      from le_antisymm (finrank_self 𝕜 ▸ l.range.finrank_le) (zero_lt_iff.mpr H),
+    have hi : function.injective (l.ker.liftq l (le_refl _)),
+    { rw [← linear_map.ker_eq_bot],
+      exact submodule.ker_liftq_eq_bot _ _ _ (le_refl _) },
+    have hs : function.surjective (l.ker.liftq l (le_refl _)),
+    { rw [← linear_map.range_eq_top, submodule.range_liftq],
+      exact eq_top_of_finrank_eq ((finrank_self 𝕜).symm ▸ this) },
+    let φ : (E ⧸ l.ker) ≃ₗ[𝕜] 𝕜 := linear_equiv.of_bijective (l.ker.liftq l (le_refl _)) hi hs,
+    have hlφ : (l : E → 𝕜) = φ ∘ l.ker.mkq,
+      by ext; refl,
+    -- Since the quotient map `E →ₗ[𝕜] (E ⧸ l.ker)` is continuous, the continuity of `l` will follow
+    -- form the continuity of `φ`.
+    suffices : continuous φ.to_equiv,
+    { rw hlφ,
+      exact this.comp continuous_quot_mk },
+    -- The pullback by `φ.symm` of the quotient topology is a T2 topology on `𝕜`, because `φ.symm`
+    -- is injective. Since `φ.symm` is linear, it is also a vector space topology.
+    -- Hence, we know that it is equal to the topology induced by the norm.
+    have : induced φ.to_equiv.symm infer_instance = hnorm.to_uniform_space.to_topological_space,
+    { refine unique_topology_of_t2 (topological_add_group_induced φ.symm.to_linear_map)
+        (has_continuous_smul_induced φ.symm.to_linear_map) _,
+      rw t2_space_iff,
+      exact λ x y hxy, @separated_by_continuous _ _ (induced _ _) _ _ _
+        continuous_induced_dom _ _ (φ.to_equiv.symm.injective.ne hxy) },
+    -- Finally, the pullback by `φ.symm` is exactly the pushforward by `φ`, so we have to prove
+    -- that `φ` is continuous when `𝕜` is endowed with the pushforward by `φ` of the quotient
+    -- topology, which is trivial by definition of the pushforward.
+    rw [this.symm, equiv.induced_symm],
+    exact continuous_coinduced_rng }
+end
+
+/-- Any linear form on a topological vector space over a nondiscrete normed field is continuous if
+    and only if its kernel is closed. -/
+lemma linear_map.continuous_iff_is_closed_ker (l : E →ₗ[𝕜] 𝕜) :
+  continuous l ↔ is_closed (l.ker : set E) :=
+⟨λ h, is_closed_singleton.preimage h, l.continuous_of_is_closed_ker⟩
+
+variables [complete_space 𝕜]
+
+/-- In finite dimension over a nondiscrete complete normed field, the canonical identification
+(in terms of a basis) with `𝕜^n` (endowed with the product topology) is continuous.
+This is the key fact wich makes all linear maps from a T2 finite dimensional TVS over such a field
+continuous (see `linear_map.continuous_of_finite_dimensional`), which in turn implies that all
+norm are equivalent in finite dimension. -/
+lemma continuous_equiv_fun_basis [ht2 : t2_space E] {ι : Type v} [fintype ι] (ξ : basis ι 𝕜 E) :
```

Comment:
I presume this is not deliberate but we shouldn't force `ι` to live in the same universe as `E`:
```suggestion
lemma continuous_equiv_fun_basis [ht2 : t2_space E] {ι : Type x} [fintype ι] (ξ : basis ι 𝕜 E) :
```

Reply:
Oh sorry I forgot about this. The first proof *does* need `ι` and `E` to be in the same universe, because of the induction step (tbh I'm not sure about the details here, the code skeleton for the induction comes from @sgouezel, and changing the universe variable just makes everything fail). But I think we could then deduce a version without such a constraint, let me try that.